src/backend/src/services/teams.services.ts

```diff
@@ -32,4 +34,47 @@ export default class TeamsService {
 
     return teamTransformer(team);
   }
+
+  /**
+   * Update the given teamId's team's members
+   * @param submitter a user who's making this request
+   * @param teamId a id of team to be updated
+   * @param userIds a array of user Ids that replaces team's old members
+   * @returns a updated team
+   * @throws if the team is not found, the submitter has no priviledge, or any user from the given userIds does not exist
+   */
+  static async updateSingleTeam(submitter: User, teamId: string, userIds: number[]): Promise<Team> {
+    // find and vertify the given teamId exist
+    const team = await prisma.team.findUnique({
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    if (!team) throw new NotFoundException('Team', teamId);
+    if (submitter.role !== Role.ADMIN && submitter.role !== Role.APP_ADMIN && submitter.userId !== team.leaderId)
+      throw new AccessDeniedException('No priviledge to update team members');
```

Comment:
the AccessDeniedException automatically says `Access Denied` so if we're adding the message then it should explain why

so maybe something like `you must be an admin or the team lead to update the members!`